### PR TITLE
Ssl connect

### PR DESCRIPTION
**Pull request by bill1600**
Handle connect failures by retrying EAGAINs and EWOULDBLOCKS a few times, then returning failure,
rather than passing an unconnected socket into SSL_connect.

### CODE REVIEW

File: src/nopoll_conn.c [hunk 1]
```diff
@@ -1004,18 +1042,20 @@ noPollConn * __nopoll_conn_new_common (noPollCtx       * ctx,
 		nopoll_log (ctx, NOPOLL_LEVEL_INFO, "connecting to remote TLS site %s:%s", conn->host, conn->port);
 		iterator = 0;
 		while (SSL_connect (conn->ssl) <= 0) {
-		
+			int log_level = NOPOLL_LEVEL_WARNING;
+			if (0 == (iterator & 0x7F))
```
Comment (shilpa24balaji) [inline src/nopoll_conn.c]: I guess these log level changes were only for test, please remove

File: src/nopoll_conn.c [hunk 2]
```diff
@@ -332,23 +375,18 @@ NOPOLL_SOCKET __nopoll_conn_sock_connect_opts_internal (noPollCtx       * ctx,
...
-			freeaddrinfo (res);
+		/* relase address info */
+		freeaddrinfo (res);
 			
-			return -1;
-		}
-	}
-	else
-	{
-		nopoll_log (ctx, NOPOLL_LEVEL_DEBUG,"socket connect successfull");
```
Comment (shilpa24balaji) [inline src/nopoll_conn.c]: Why is this deleted?

**Review by shilpa24balaji — CHANGES_REQUESTED** [review]

File: src/nopoll_conn.c [hunk 3]
```diff
@@ -882,7 +926,7 @@ noPollConn * __nopoll_conn_new_common (noPollCtx       * ctx,
 	if (session == NOPOLL_INVALID_SOCKET) {
 		/* release connection options */
 		__nopoll_conn_opts_release_if_needed (options);
-		nopoll_log (ctx, NOPOLL_LEVEL_DEBUG, "Failed to connect to remote host %s:%s", host_ip, host_port);
+		nopoll_log (ctx, NOPOLL_LEVEL_CRITICAL, "Failed to connect to remote host %s:%s", host_ip, host_port);
```
Comment (shilpa24balaji) [inline src/nopoll_conn.c]: @bill1600 Is this log level change needed?
We have other logs to indicate the connection host and port.

Comment (bill1600) [reply src/nopoll_conn.c]: Don't we want to see a log message when connection has failed ?

Comment (bill1600) [reply src/nopoll_conn.c]: Don' we want to see a log message when the connection fails ?

Comment (shilpa24balaji) [reply src/nopoll_conn.c]: No this log does not give any extra information. All that info is already known in the caller i.e. parodus

**Review by shilpa24balaji — APPROVED** [review]
